drivers/sensor/iis2dlpc/iis2dlpc_trigger.c

```diff
@@ -29,46 +29,47 @@ static int iis2dlpc_enable_int(const struct device *dev,
 	struct iis2dlpc_data *iis2dlpc = dev->data;
 	iis2dlpc_reg_t int_route;
 
-	if (cfg->drdy_int == 1U) {
+	switch (type) {
+	case SENSOR_TRIG_DATA_READY:
+		if (cfg->drdy_int == 1) {
+			/* set interrupt for pin INT1 */
+			iis2dlpc_pin_int1_route_get(iis2dlpc->ctx,
+					&int_route.ctrl4_int1_pad_ctrl);
+			int_route.ctrl4_int1_pad_ctrl.int1_drdy = enable;
+
+			return iis2dlpc_pin_int1_route_set(iis2dlpc->ctx,
+					&int_route.ctrl4_int1_pad_ctrl);
+		} else {
+			/* set interrupt for pin INT2 */
+			iis2dlpc_pin_int2_route_get(iis2dlpc->ctx,
+					&int_route.ctrl5_int2_pad_ctrl);
+			int_route.ctrl5_int2_pad_ctrl.int2_drdy = enable;
+
+			return iis2dlpc_pin_int2_route_set(iis2dlpc->ctx,
+					&int_route.ctrl5_int2_pad_ctrl);
+		}
+		break;
```

Comment:
It looks like you're mixing drdy_int changes with tap/pulse changes? Can you move this section to the commit `drivers/sensor: iis2dlpc: Move drdy_int info into DT`

Reply:
> It looks like you're mixing drdy_int changes with tap/pulse changes? Can you move this section to the commit `drivers/sensor: iis2dlpc: Move drdy_int info into DT`

mmmmh, actually it is neither belonging to the first one nor to the second one. It is just a different way to handle the switch of trigger types. But you are right in saying that does no9t belong to the TAP changes. So, if you agree, I can move this change to an intermediate commit.

Reply:
Yes, please

Reply:
OK, splitted the work into 3 commits. The intermediate one is carrying the "switch change", which is done just for a better readability.